packages/ui/src/components/TreeView/TreeView.tsx

```diff
@@ -0,0 +1,224 @@
+          onEditSubmit?.(localValueState)
+        }
+      }
+
+      document.addEventListener('mousedown', handleClickOutside)
+
+      return () => {
+        document.removeEventListener('mousedown', handleClickOutside)
+      }
+    }, [localValueState])
+
+    useEffect(() => {
+      if (isEditing) {
+        console.log('useEffect with inputRef.current?.focus()')
+        inputRef.current?.focus()
+        // formRef.current?.focus()
+      }
+    }, [isEditing])
+
+    const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+      e.preventDefault()
+      onEditSubmit?.(localValueState)
+    }
+
+    return (
+      <div
+        ref={ref}
+        aria-selected={isSelected}
+        aria-expanded={isExpanded}
+        {...props}
+        className={cn(
+          'relative',
+          'transition-colors',
+          'flex items-center gap-3',
+          'text-sm',
+          'cursor-pointer',
+          'select-none',
+          'text-foreground-light',
+          'aria-selected:text-foreground',
+          'aria-expanded:bg-control',
+          'aria-selected:!bg-selection',
+          'data-[state=open]:bg-control', // bg state for context menu open
+          'group',
+          'h-[28px]',
+          'hover:bg-control'
+        )}
+        style={{
+          paddingLeft:
+            level === 1 && !isBranch
+              ? xPadding + levelPadding / 2 - 2
+              : level
+                ? levelPadding * (level - 1) + xPadding + (!isBranch ? 0 : 0)
+                : levelPadding,
+          ...props.style,
+        }}
+        data-treeview-is-branch={isBranch}
+        data-treeview-level={level}
+      >
+        {level && level > 1 && (
+          <div
+            style={{
+              left: (levelPadding / 2 + 4) * (level - 1) + xPadding,
+            }}
+            className={
+              'absolute h-full w-px group-data-[treeview-is-branch=false]:bg-border-strong'
+            }
+          ></div>
+        )}
+        <div className="absolute left-0 h-full w-0.5 group-aria-selected:bg-foreground"></div>
+        {isBranch ? (
+          <>
+            <ChevronRight
+              className={cn(
+                'text-foreground-muted',
+                'group-aria-selected:text-foreground-light',
+                'group-aria-expanded:text-foreground-light',
+                'transition-transform duration-200',
+                'group-aria-expanded:rotate-90'
+              )}
+              size={14}
+            />
+            <TreeViewFolderIcon
+              className={cn(
+                'transition-colors',
+                ' text-foreground-muted',
+                'group-aria-selected:text-foreground-light',
+                'group-aria-expanded:text-foreground-light'
+              )}
+              isOpen={isExpanded}
+              size={16}
+              strokeWidth={1.5}
+            />
+          </>
+        ) : (
+          <SQL_ICON
```

Comment:
should we have the icon be passed in as a prop? make it extensible
can default to SQL_ICON if none provided

Reply:
yea sure

Reply:
still pending to change by me